examples/llama.py

```diff
@@ -252,7 +261,21 @@ def convert_from_huggingface(weights, model):
     "model.norm.weight": "norm.weight",
     "lm_head.weight": "output.weight",
   }
-  return {keymap[k]: v for k,v in weights.items() if ".rotary_emb." not in k}
+  sd = {}
+  for k, v in weights.items():
+    if ".rotary_emb." in k:
+      continue
+    v = v.to(Device.DEFAULT)
+    to_name = keymap[k]
+    if "model.layers" in k:
+      if "q_proj" in k:
+        sd[to_name] = permute(v, 32, 2048, 2048)
```

Comment:
use the params instead of hard coding the numbers

Reply:
done